games/devtest/mods/unittests/async_env.lua

```diff
@@ -166,3 +166,17 @@ local function test_userdata_passing2(cb, _, pos)
 	end, vm, pos)
 end
 unittests.register("test_userdata_passing2", test_userdata_passing2, {map=true, async=true})
+
+local function test_vector_preserve(cb)
+	local vec = vector.new(1, 2, 3)
+	core.handle_async(function(x)
+		return x
+	end, function(ret)
+		if ret == vec then
```

Comment:
```suggestion
		if ret == vec then -- only succeeds if metatable was preserved
```
could alternatively check directly using getmetatable

Reply:
I would go for the current approach as it also makes sure that the code for setting the metatable doesn't (somehow) mess up the content of the table.